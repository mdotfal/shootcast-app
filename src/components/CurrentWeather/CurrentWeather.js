import React from 'react';
import './CurrentWeather.css';

function CurrentWeather( props ) {

  const sunset = ( stamp ) => {
    const timeFormat = new Intl.DateTimeFormat( 'en-US', {
      timeStyle: 'medium',
      timeZone: 'America/Los_Angeles'
    });
    return timeFormat.format( new Date( stamp * 1000 ))
  }

  const { name } = props.wData;
  return (
    <div className='current-weather'>
      { props.wData.length !== 0
          ? <div>
              <p><strong>{ props.city }</strong></p>
              <h1>{ Math.round( props.wData.main.temp ) }°F</h1>
              <p>{ props.wData.weather[0].description }</p>
              <p style={{ color: `rgb(255, 181, 43)` }}> Sunset: { sunset( props.wData.sys.sunset ) }</p>
              <p style={{ color: `rgb(150, 72, 0)` }}> Goldenhour: { sunset( props.wData.sys.sunset - 3600 )}</p>
            </div>
          : <h1> { name } </h1>
      }
    </div>
  );
};

export default CurrentWeather;