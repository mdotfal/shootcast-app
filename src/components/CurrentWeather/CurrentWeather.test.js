import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CurrentWeather from './CurrentWeather';

describe( `CurrentWeather component`, () => {
  it(Â `renders without errors`, () => {
    const wData = {
      name: "Test name",
      main: [
        { temp: 50 }
      ],
      weather: [
        { description: "test description" }
      ],
      sys: {
        sunset: "123456"
      }
    };
    const wrapper = shallow( <CurrentWeather wData = { wData }/> );
    expect( toJson( wrapper )).toMatchSnapshot();
  });
});