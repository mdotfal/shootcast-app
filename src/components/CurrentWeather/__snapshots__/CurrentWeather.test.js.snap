// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentWeather component renders without errors 1`] = `
<div
  className="current-weather"
>
  <div>
    <p>
      <strong />
    </p>
    <h1>
      NaN
      Â°F
    </h1>
    <p>
      test description
    </p>
    <p
      style={
        Object {
          "color": "rgb(255, 181, 43)",
        }
      }
    >
       Sunset: 
      2:17:36 AM
    </p>
    <p
      style={
        Object {
          "color": "rgb(150, 72, 0)",
        }
      }
    >
       Goldenhour: 
      1:17:36 AM
    </p>
  </div>
</div>
`;
